importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -2455,6 +2470,38 @@ QString MusicXMLParserDirection::placement() const
       else return _placement;
       }
 
+//---------------------------------------------------------
+//   combineTempoText
+//---------------------------------------------------------
+/**
+ Combine potentially separated tempo text.
+ */
+
+void DelayedDirectionsList::combineTempoText()
+      {
+      // Iterate through candidates 
+      for (auto ddi1 = rbegin(), ddi1Next = ddi1; ddi1 != rend(); ddi1 = ddi1Next) {
+            ddi1Next = std::next(ddi1);
+            if ((*ddi1)->isTempoOrphanCandidate()) {
+                  for (auto ddi2 = rbegin(), ddi2Next = ddi2; ddi2 != rend(); ddi2 = ddi2Next) {
+                        ddi2Next = std::next(ddi2);
+                        // Combine with tempo text if present
+                        if (ddi1 != ddi2
+                            && (*ddi2)->tick() == (*ddi1)->tick()
+                            && (*ddi2)->element()->isTempoText()) {
+                              TempoText* tt = toTempoText((*ddi2)->element());
+                              StaffText* st = toStaffText((*ddi1)->element());
+                              QString sep = tt->plainText().endsWith(' ') || st->plainText().startsWith(' ') ? "" : " ";
+                              tt->setXmlText(tt->xmlText() + sep + st->xmlText());
+                              delete st;
```

Comment:
is just deleting the StaffText like that safe?  I'd expect to have to call "removeElement" on its parent...

Reply:
Safe! At this point in the code the elements have not yet been added to the score, so they don't yet have parents.